sys/riotboot/flashwrite.c

```diff
@@ -150,6 +151,33 @@ int riotboot_flashwrite_putbytes(riotboot_flashwrite_t *state,
     return 0;
 }
 
+int riotboot_flashwrite_invalidate(int slot)
+{
+    if (riotboot_slot_numof == 1){
+        LOG_WARNING(LOG_PREFIX "abort, only one slot configured\n");
+        return -1;
+    }
+    if (riotboot_slot_validate(1 - slot) != 0){
+        LOG_WARNING(LOG_PREFIX "abort, slot %d is the only valid image\n", slot);
+        return -2;
+    }
```

Comment:
Both could be invalid though, so this statement is not necessarily accurate. Better phrase the other way around 
`abort, cant erase slot %d, other slot is not a valid, , slot`.